### DISCUSSION TITLE
`ThemeVariant`-compliance problem for custom theme

### DISCUSSION THREAD

**Discussion by Splitwirez** [AvaloniaUI/Avalonia#10451] [Q&A]
Context: As of Avalonia 11.0.0-preview5, the built-in `FluentTheme` no longer has its `Mode` property to toggle between dark and light variants - in its place, some new `ThemeVariant` infrstructure exists, which is set at the `App`/`TopLevel`/`Window`/`ThemevariantScope` level, instead of on the theme itself.

Whilst updating my theme (Aero for Avalonia) to 11.0.0-preview5, I figured I'd best follow that precedent...but I've had some trouble doing so thus far.

Here's `AeroTheme.axaml`:
```xml
<Styles xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        x:Class="AvaloniaAero.AeroTheme">
    <Styles.Resources>
        <ResourceDictionary>
            <ResourceDictionary.ThemeDictionaries>
                <!-- Light variant includes (light should be the "default", yes?) -->
                <ResourceDictionary x:Key="Default">
                    <ResourceDictionary.MergedDictionaries>
                        <ResourceInclude Source="avares://AvaloniaAero/AVMAP/All.axaml" />
                        <ResourceInclude Source="avares://AvaloniaAero/Variant/Light.axaml" />
                    </ResourceDictionary.MergedDictionaries>
                </ResourceDictionary>

                <!-- Epic include fail -->
                <!--
                <ResourceDictionary x:Key="Light">
                    <ResourceDictionary.MergedDictionaries>
                        <ResourceInclude Source="avares://AvaloniaAero/AVMAP/All.axaml" />
                        <ResourceInclude Source="avares://AvaloniaAero/Variant/Light.axaml" />
                    </ResourceDictionary.MergedDictionaries>
                </ResourceDictionary>
                -->

                <!-- Dark variant includes -->
                <ResourceDictionary x:Key="Dark">
                    <ResourceDictionary.MergedDictionaries>
                        <ResourceInclude Source="avares://AvaloniaAero/AVMAP/All.axaml" />
                        <ResourceInclude Source="avares://AvaloniaAero/Variant/Dark.axaml" />
                    </ResourceDictionary.MergedDictionaries>
                </ResourceDictionary>
            </ResourceDictionary.ThemeDictionaries>
        </ResourceDictionary>
    </Styles.Resources>
    
    <!-- Styles included here -->
    <StyleInclude Source="avares://AvaloniaAero/AVMAP/StyleTrailMix.axaml" />
    <StyleInclude Source="avares://AvaloniaAero/AVMAP/comctlAVA.axaml" />
</Styles>
```

The rest of the code is at https://github.com/Splitwirez/avalonia-aero-theme/tree/migration/avalonia--11-0-0-preview5 if needed. Problem is, no matter what I do, the dark variant includes never seem to be applied, only the default/light variant includes.

I've tried:

	- Specifying a theme dictionary with `x:Key="Light"`, in place of the `x:Key="Default"` one
	- Specifying a theme dictionary with `x:Key="Light"`, in addition to the `x:Key="Default"` one

...to no avail. As a diagnostic measure, I also tried:

	- Setting `RequestedThemeVariant="Dark"` in the accompanying demo's `App.axaml`
	- Setting `Application.Current.RequestedThemeVariant = ThemeVariant.Dark;` in code in the accompanying demo
	- Setting `MainWindow.RequestedThemeVariant = ThemeVariant.Dark;` (there is only one window in the accompanying demo) in code

...once more, no change. (for context, I'm on Kubuntu Linux 20.04 x64)

At this point, I feel like I must've misunderstood how this `ThemeVariant` stuff is supposed to work or something. What do?

**Comment by maxkatz6** [comment]
Had another look at this problem, and it was finally fixed by this PR https://github.com/AvaloniaUI/Avalonia/pull/11111
